### PR TITLE
i593: add support for number ranges in yaml

### LINKED ISSUE #593: Add support for integer ranges in yaml and other fields like auto judge number field
**Is your feature request related to a problem?** 

No.   

For all yaml fields where a list of comma delimited numbers is allowed, add support for range of numbers

When entering number: 12,13,14,15
add support for ranges, thus, number: 12-15
support number list and ranges, thus: : 1,4,12-15,22   would expand to 1, 4, 12, 13, 14, 15, 22

**Feature Description**: 

For multiple integer field values allow a range (#-##) instead
of having to create a list of comma delimited list of integers.
allows spaces in number string, trim spaces
Do not allow invalid ranges where end number is greater than end number, ex 23-12
Do not allow invalid range, missing end or start number,  ex -4 or 55-
Require all range numbers are greater than zero.

**Have you considered other ways to accomplish the same thing?** 

Work around is to add a long list of comma delimited integers

**Do you have any specific suggestions for how your feature would be ***implemented*** in PC^2?**

For any yaml field, or other text entry field.  Create a static method and put it into a utility class.
in StringUtilities, add  public static int[] getNumberList(String numberString)

**Additional context**:

### PR DESCRIPTION
**Pull request by lane55**
Now loads number list with using StringUtilities.getNumberList which now supports ranges of values.

### Description of what the PR does

Add support for ranges where number lists are used in yaml 

Including yaml fields:
auto-judging number
team-proxy-accounts number

### Issue which the PR fixes

Fixes #593

### Environment in which the PR was developed (OS,IDE, Java version, etc.)

Microsoft Windows [Version 10.0.19044.2130]

java version "1.8.0_121"
Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.121-b13, mixed mode)

### Precise steps for _testing_ the PR (i.e., how to demonstrate that it works correctly)

Add ranges into a yaml 

In mini contest change auto judge number line in contest.yaml
from 
    number: 7,8,9,10,11,12,13,14
to
    number: 7,8-10,14

load mini contest

Expected

Auto judges 7,8,9,10 and 14 should be assigned problem A

### CODE REVIEW

**Comment by troy2914** [comment]
Please rebase this,  it has conflicts.

**Comment by lane55** [comment]
Merged with develop, no conflicts with the base branch

File: src/edu/csus/ecs/pc2/core/StringUtilities.java [hunk 1]
```diff
@@ -277,7 +278,87 @@ public static boolean getBooleanValue(String string, boolean defaultBoolean) {
     }
     
     /**
-     * get team number from login string.
+<<<<<<< HEAD
```
Comment (troy2914) [inline src/edu/csus/ecs/pc2/core/StringUtilities.java:281]: that is a conflict marker, please remove it.

File: src/edu/csus/ecs/pc2/imports/ccs/ContestSnakeYAMLLoader.java [hunk 2]
```diff
@@ -2258,38 +2258,13 @@ protected int toInt(String string, int defaultNumber) {
...
-
-        int number = defaultNumber;
-
-        if (string != null && string.length() > 0) {
-            number = Integer.parseInt(string.trim());
-        }
-
-        return number;
+        // TODO REFACTOR replace getIntegerValue with StringUtilities.getIntegerValue
+        return StringUtilities.getIntegerValue(string, defaultNumber);
```
Comment (troy2914) [inline src/edu/csus/ecs/pc2/imports/ccs/ContestSnakeYAMLLoader.java:2262]: seems that TODO todone?   so get rid of the TODO

File: src/edu/csus/ecs/pc2/imports/ccs/ContestSnakeYAMLLoader.java [hunk 3]
```diff
@@ -2258,38 +2258,13 @@ protected int toInt(String string, int defaultNumber) {
...
-                out[i] = getIntegerValue(n, 0);
-                // if (out[i] < 1) {
-                // // SOMEDAY 669 throw invalid number in list exception
-                // }
-                i++;
-            }
-            return out;
-        }
+        // TODO REFACTOR replace getNumberList with StringUtilities.getNumberList
+        return StringUtilities.getNumberList(numberString);
```
Comment (troy2914) [inline src/edu/csus/ecs/pc2/imports/ccs/ContestSnakeYAMLLoader.java:2267]: same this with TODO as with the last item.

**Review by troy2914 — CHANGES_REQUESTED** [review]
I tested the code and it worked as expected.   Unfortunately there crap things in the code in comments.

File: src/edu/csus/ecs/pc2/imports/ccs/ContestSnakeYAMLLoader.java (hunk 2 above)
Comment (lane55) [reply src/edu/csus/ecs/pc2/imports/ccs/ContestSnakeYAMLLoader.java:2262]: willd o

Comment (lane55) [reply src/edu/csus/ecs/pc2/imports/ccs/ContestSnakeYAMLLoader.java]: pushed fix

**Review by troy2914 — CHANGES_REQUESTED** [review]
still that <<<<< HEAD conflict marker that needs to cleaned up.

**Review by troy2914 — APPROVED** [review]
Thanks, looks good now.

**Pull request merged**